dotnet/samples/Concepts/Optimization/PluginSelection.cs

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.ComponentModel;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+
+namespace Optimization;
+
+/// <summary>
+/// Single kernel instance may have multiple imported plugins/functions. It's possible to enable automatic function calling,
+/// so AI model will decide which functions to call for specific request.
+/// In case there are a lot of plugins/functions in application, some of them (or all of them) need to be shared with the model.
+/// This example shows how to use different plugin/function selection strategies, to share with AI only those functions,
+/// which are related to specific request.
+/// This technique should decrease token usage, as fewer functions will be shared with AI.
```

Comment:
We may also want to mention the enterprise scenario here, where someone is building a general purpose chat experience for a large enterprise, where they have so many plugins, that it's impossible to share all with the LLM.